packages/yarnpkg-core/tests/semverUtils.test.ts

```diff
@@ -1,27 +1,68 @@
+import semver           from 'semver';
+
 import * as semverUtils from '../sources/semverUtils';
 
-const SPECS: Array<[string, string, boolean]> = [
+type Specs = Array<[string, string, boolean]>;
+
+const SPECS: Specs = [
   // Those three are different from includePrerelease
   [`^5.0.0`, `5.0.0-beta.13`, true],
   [`^1.0.0`, `1.0.0-rc1`, true],
   [`2.x`, `3.0.0-pre.0`, false],
 
-  [`2.x`, `2.0.0-pre.0`, true],
-  [`2.x`, `2.1.0-pre.0`, true],
-  [`*`, `1.0.0-rc1`, true],
-  [`^1.0.0-0`, `1.0.1-rc1`, true],
-  [`^1.0.0-rc2`, `1.0.1-rc1`, true],
-  [`^1.0.0`, `1.0.1-rc1`, true],
-  [`^1.0.0`, `1.1.0-rc1`, true],
+  // From the semver test-suite
+  ...[
+    [`2.x`, `2.0.0-pre.0`, true],
+    [`2.x`, `2.1.0-pre.0`, true],
+    [`1.1.x`, `1.1.0-a`, true],
+    [`1.1.x`, `1.1.1-a`, true],
+    [`*`, `1.0.0-rc1`, true],
+    [`^1.0.0-0`, `1.0.1-rc1`, true],
+    [`^1.0.0-rc2`, `1.0.1-rc1`, true],
```

Comment:
I'm really not sure about this one. We use `satisfiesWithPrereleases` to allow users to use prereleases even when the range doesn't specify one - not to allow to ignore the one that might exist.

What I mean by that is that in this case, since the one who described the peer dependency took the care to explicitly specify a prerelease lower bound, it makes sense to assume that lower prereleases aren't supported at all (by contrast, when the prerelease lower bound is missing, it makes sense to assume that it's just an omission).

So returning `true` on `1.0.1-rc1` matching `^1.0.0-rc2` feels like a bug to me 🤔

Reply:
🤔 I think you're right. I'll leave those commented out until we do something about it (it's not affected by this PR, it's just part of the semver test suite).

Reply:
But it means that if we merge this PR as-is we'll introduce a new bug, right?

Reply:
Nope, that test has the same behavior both on master and in this PR. It's just that I've copied it from the test suite of the `semver` repository (https://github.com/npm/node-semver/blob/60880700d6cee8450e37f5ac31430d84587620f5/test/fixtures/range-include.js#L116) along with the other tests. I've commented it out now, so that we can adress it in the future.